app/workers/check_for_new_commits_on_github_worker.rb

```diff
@@ -14,32 +16,49 @@ class CheckForNewCommitsOnGithubWorker < GitHubWorkerBase
 
     attr_accessor :trigger_type
     attr_accessor :scheduler
+    attr_reader :github_service
 
     def initialize(provider_credential: nil, project: nil)
       self.trigger_type = FastlaneCI::JobTrigger::TRIGGER_TYPE[:commit]
       self.scheduler = WorkerScheduler.new(interval_time: 10)
+      @github_service = FastlaneCI::GitHubService.new(provider_credential: provider_credential)
 
       super(provider_credential: provider_credential, project: project) # This starts the work by calling `work`
     end
 
-    def work
-      logger.debug("Checking for new commits in #{self.project.project_name}")
-      repo = self.git_repo
-
-      # TODO: ensure BuildService subclasses are thread-safe
+    def check_for_new_commits
```

Comment:
this is much cleaner if you look at the whole file instead of this diff. It's very understandable